R/derive_var_extreme_flag.R

```diff
@@ -196,53 +131,190 @@
+#'   date and then select the last one (`mode = "last"`). The name of the new
+#'   variable is passed through the `new_var = LASTFL` call.
+#'
+#' @code
+#' advs %>%
+#'   derive_var_extreme_flag(
+#'     by_vars = exprs(STUDYID, USUBJID, PARAMCD),
+#'     order = exprs(ADT),
+#'     new_var = LASTFL,
+#'     mode = "last",
+#'   ) %>%
+#'   arrange(STUDYID, USUBJID, PARAMCD, ADT) %>%
+#'   select(STUDYID, everything())
 #'
-#' # Most severe AE first occurrence per patient
-#' example_ae %>%
+#' @info Note here that a similar `FIRSTFL` variable could instead be derived
+#'    simply by switching to `mode = "first"`. Alternatively, we could make use
+#'    of `desc()` within the sorting specified by `order`:
+#'
+#' @code
+#' advs %>%
+#'   derive_var_extreme_flag(
+#'     by_vars = exprs(STUDYID, USUBJID, PARAMCD),
+#'     order = exprs(desc(ADT)),
+#'     new_var = FIRSTFL,
+#'     mode = "last",
+#'   ) %>%
+#'   arrange(STUDYID, USUBJID, PARAMCD, ADT) %>%
+#'   select(STUDYID, everything())
+#'
+#' @caption Modifying the flag values (`true_value`, `false_value`)
+#'
+#' @info The previous example is now enhanced with custom values for the flag
+#'   entries. Records which are flagged are filled with the contents of
+#'   `true_value` and those which are not are filled with the contents of
+#'   `false_value`. Note that these are normally preset to `"Y"` and `NA`,
+#'   which is why they were not specified in the example above.
+#'
+#' @code
+#' advs %>%
+#'   derive_var_extreme_flag(
+#'     by_vars = exprs(STUDYID, USUBJID, PARAMCD),
+#'     order = exprs(ADT),
+#'     new_var = LASTFL,
+#'     mode = "last",
+#'     true_value = "Yes",
+#'     false_value = "No",
+#'   ) %>%
+#'   arrange(STUDYID, USUBJID, PARAMCD, ADT) %>%
+#'   select(STUDYID, everything())
+#'
+#' @caption Creating temporary variables for sorting
+#'
+#' @info In this example we wish to flag the first occurrence of the most
+#'   severe AE within each subject. To ensure correct sorting of the
+#'   severity values, `AESEV` must be pre-processed into a numeric variable
+#'   `TEMP_AESEVN` which can then be passed inside `order`. Once again,
+#'   to ensure we only flag the *first* occurrence, we specify `AESDTY` and
+#'   `AESEQ` inside `order` as well.
+#'
+#' @code
+#' adae %>%
 #'   mutate(
 #'     TEMP_AESEVN =
 #'       as.integer(factor(AESEV, levels = c("SEVERE", "MODERATE", "MILD")))
 #'   ) %>%
 #'   derive_var_extreme_flag(
 #'     new_var = AOCCIFL,
-#'     by_vars = exprs(USUBJID),
+#'     by_vars = exprs(STUDYID, USUBJID),
 #'     order = exprs(TEMP_AESEVN, AESTDY, AESEQ),
 #'     mode = "first"
 #'   ) %>%
-#'   arrange(USUBJID, AESTDY, AESEQ) %>%
-#'   select(USUBJID, AEDECOD, AESEV, AESTDY, AESEQ, AOCCIFL)
+#'   arrange(STUDYID, USUBJID, AESTDY, AESEQ) %>%
+#'   select(STUDYID, USUBJID, AEDECOD, AESEV, AESTDY, AESEQ, AOCCIFL)
+#'
+#' @caption Flagging all records if multiple are identified (flag_all`)
```

Comment:
just spotted this, will update after reviews come in 😄 

```suggestion
#' @caption Flagging all records if multiple are identified (`flag_all`)
```